.github/workflows/npm-screens-publish-nightly.yml

```diff
@@ -0,0 +1,65 @@
+name: Build nightly and publish on npm
+env:
+  YARN_ENABLE_HARDENED_MODE: 0
+on:
+  schedule:
+    - cron: '27 23 * * *' # at 23:27 every day
+jobs:
+  npm-build:
+    if: github.repository == 'software-mansion/react-native-screens'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+    env:
+      PACKAGE_NAME: PLACEHOLDER # Will be reassigned later on.
+    steps:
+      - name: Check out
+        uses: actions/checkout@v4
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: 'yarn'
+          registry-url: https://registry.npmjs.org/
+
+      - name: Clear annotations
+        run: .github/workflows/helper/clear-annotations.sh
+
+      - name: Install node dependencies
+        run: yarn install --immutable
+
+      - name: Build package
+        id: build
+        run: ./createNPMPackage.sh
+
+      - name: Check if any node_modules were packed
+        id: node_modules
+        run: ! grep --silent -E "node_modules/.+" build.log
+
+      - name: Show build log
+        if: failure() && steps.build.outcome == 'failure'
+        run: cat build.log
+
+      - name: Show packed node_modules
+        if: failure() && steps.node_modules.outcome == 'failure'
+        run: ! grep -E "node_modules/.+" build.log
+
+      - name: Add package name to env
+        run: echo "PACKAGE_NAME=$(ls -l | egrep -o "react-native-screens-(.*)(=?\.tgz)")" >> $GITHUB_ENV
+
+      - name: Assert PACKAGE_NAME
+        if: ${{ env.PACKAGE_NAME == 'PLACEHOLDER' }}
+        run: exit 1 # This should never happen.
+
+      - name: Upload npm package to GitHub
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ env.PACKAGE_NAME }}
+          path: './${{ env.PACKAGE_NAME }}'
+
+      - name: Publish npm package
+        run: npm publish $PACKAGE_NAME --tag nightly --provenance
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NODE_AUTH_TOKEN }}
```

Comment:
Where does this secret come from? I guess we have to generate it & put into GitHub secrets system, right?

Reply:
You need to generate a npm token on npm website, it can be either yours or we can create `softwaremansion` account on npm for this case.

Reply:
Yes, that's the action repository secret. And the value is npm access token. I think we can use our "official" npm user and not the "dummy" one.